core/src/main/java/io/parsingdata/metal/expression/value/Expand.java

```diff
@@ -56,11 +58,11 @@ public Expand(final ValueExpression bases, final ValueExpression count) {
         if (baseList.isEmpty()) {
             return baseList;
         }
-        final ImmutableList<Value> countList = count.eval(parseState, encoding);
-        if (countList.size != 1 || countList.head.equals(NOT_A_VALUE)) {
-            throw new IllegalArgumentException("Count must evaluate to a single non-empty value.");
+        final Optional<Value> countValue = count.evalSingle(parseState, encoding);
+        if (!countValue.isPresent() || countValue.get().equals(NOT_A_VALUE)) {
+            throw new IllegalArgumentException("Count must evaluate to a non-empty countable value.");
         }
-        return expand(baseList, countList.head.asNumeric().intValueExact(), new ImmutableList<>()).computeResult();
+        return expand(baseList, countValue.get().asNumeric().intValueExact(), new ImmutableList<>()).computeResult();
```

Comment:
Is it possible to do `return countValue.filter(!NOT_A_VALUE).map(expand(...)).orElseThrow()` here?

Reply:
Wow indeed!